overlord/snapstate/snapstate.go

```diff
@@ -1619,11 +1619,16 @@ func canRemove(st *state.State, si *snap.Info, snapst *SnapState, removeAll bool
 		if model.Base() != "" && !coreInUse(st) {
 			return true
 		}
+		return false
 	}
 
-	// You never want to remove a kernel or OS. Do not remove their
-	// last revision left.
-	if si.Type == snap.TypeKernel || si.Type == snap.TypeOS {
+	// Because of a Remodel() we may have multiple kernels. Allow
+	// removals of kernel(s) that are not model kernels (anymore).
+	if si.Type == snap.TypeKernel {
+		model := deviceCtx.Model()
+		if model.Kernel() != si.InstanceName() {
```

Comment:
This can be abbreviated to just
```
return model.Kernel() != si.InstanceName()
```

Reply:
+1 to zyga's suggestion

Reply:
Good point, the check above (for OS) is equally "verbose" and written in this style this is why its here. I slightly prefer it to be consistent. But no strong opinions if you prefer the other style we can do a followup which changes both places :)

Reply:
There's also two model := ... now, that could be lifted out.

I also wonder whether in this new world `boot.InUse(si.InstanceName(), si.Revision)` should come before this checks, to make sure we don't remove a kernel before it's been switched out of boot env too.